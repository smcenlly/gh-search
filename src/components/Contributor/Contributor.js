import React from "react";
import i18n from "i18next";
import styled from "styled-components";
import PropTypes from "prop-types";

const ContributorItem = styled.li`
    & + & {
        margin-top: 12px;
    }
`;

class Contributor extends React.PureComponent {
    static propTypes = {
        login: PropTypes.string,
        contributions: PropTypes.number
    };
    render() {
        let { className, login, contributions } = this.props;
        return (
            <ContributorItem className={className}>
                {i18n.t("contributor.contributions", {
                    login,
                    count: contributions
                })}
            </ContributorItem>
        );
    }
}

export default Contributor;
