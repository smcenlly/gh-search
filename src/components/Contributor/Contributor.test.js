import React from "react";
import { shallow } from "enzyme";
import Contributor from "./Contributor";

describe("Contributor", function() {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Contributor />);
    });

    it("renders without crashing", () => {
        expect(wrapper).toExist();
    });

    it("matches snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
