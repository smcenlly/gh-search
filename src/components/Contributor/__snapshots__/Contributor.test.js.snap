// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributor matches snapshot 1`] = `<styled.li />`;
